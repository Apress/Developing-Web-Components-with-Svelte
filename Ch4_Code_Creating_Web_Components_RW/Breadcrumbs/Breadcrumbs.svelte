<svelte:options tag="cobalt-breadcrumbs" />

<script>
  import arrow from "./icons/arrow-forward-outline.svg";

  export let divider = "/";
  export let image = false;
  // export let breadcrumbItems = [
  //   { href: "/", text: "Dashboard" },
  //   { href: "/reports", text: "Annual reports" },
  //   { href: "/reports/2019", text: "2019" },
  // ];
  import breadcrumbItemsData from "./breadcrumbsdata.json";

  export let breadcrumbItems = [];

  if (breadcrumbItems == []) {
    breadcrumbItems = breadcrumbItemsData;
  }
</script>

<div class="cobalt">
  <ul class="breadcrumb">
    {#each breadcrumbItems as breadcrumbItem, i}
      <li>
        <!-- Breadcrumb divider -->
        {#if i !== 0}
          {#if !image}
            <span>{divider}</span>
          {/if}

          {#if image}
            <img src={arrow} alt="arrow" height="15" width="15" />
          {/if}
        {/if}

        <!-- Render each breadcrumb -->
        {#if i === breadcrumbItems.length - 1}
          {breadcrumbItem.text}
        {:else}
          <a href={breadcrumbItem.href}>{breadcrumbItem.text}</a>
        {/if}
      </li>
    {/each}
  </ul>
</div>

<style>
  .cobalt {
    display: flex;
    font-family: Arial, Helvetica, sans-serif;
  }

  ul.breadcrumb {
    padding: 10px 16px;
    list-style: none;
    background-color: #eee;
  }

  ul.breadcrumb li {
    display: inline;
    font-size: 18px;
  }

  ul.breadcrumb li a {
    color: #19247c;
    text-decoration: none;
  }

  ul.breadcrumb li a:hover {
    color: #9393ff;
    text-decoration: underline;
  }

  ul.breadcrumb li span {
    display: inline;
    padding: 8px;
  }
</style>
