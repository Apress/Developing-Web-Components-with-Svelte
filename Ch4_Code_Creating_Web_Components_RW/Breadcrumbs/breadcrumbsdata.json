[
  { "href": "/", "text": "Dashboard" },
  { "href": "/reports", "text": "Annual reports" },
  { "href": "/reports/2019", "text": "2019" }
]