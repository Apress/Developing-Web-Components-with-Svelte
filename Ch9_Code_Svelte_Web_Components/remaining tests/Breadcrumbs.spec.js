import { render, screen } from '@testing-library/svelte';
import { fireEvent } from '@testing-library/dom';
import Breadcrumbs from "../src/lib/Breadcrumbs/Breadcrumbs.svelte";

describe("Tests for Breadcrumbs", () => {
  const props = { 
    divider: "/",
    image: false,
    breadcrumbItems: [
      { href: "/", text: "Dashboard" },
      { href: "/reports", text: "Annual reports" },
      { href: "/reports/2019", text: "2019" },
    ],
  };

  it("should render properly", async() => {
    const result = render(Breadcrumbs, { props });
    expect(() => result).not.toThrow();
  });

  it("should show the Breadcrumbs when clicking the button", async() => {
    const result = render(Breadcrumbs, { props });
    const links = result.getAllByTestId("breadcrumbLink");
    expect(links.length).toBe(2);
  })

  it("get a snapshot of component", () => {
    const tree = render(Breadcrumbs, props);
    expect(tree).toMatchSnapshot();
  })
});