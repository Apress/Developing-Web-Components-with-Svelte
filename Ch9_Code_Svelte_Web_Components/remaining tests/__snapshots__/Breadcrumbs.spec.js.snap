// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Breadcrumbs get a snapshot of component 1`] = `
{
  "component": Breadcrumbs {
    "$$": {
      "after_update": [],
      "before_update": [],
      "bound": {},
      "callbacks": {},
      "context": Map {},
      "ctx": [
        [
          {
            "href": "/",
            "text": "Dashboard",
          },
          {
            "href": "/reports",
            "text": "Annual reports",
          },
          {
            "href": "/reports/2019",
            "text": "2019",
          },
        ],
        "/",
        false,
      ],
      "dirty": [
        -1,
      ],
      "fragment": {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": [
        [Function],
      ],
      "on_disconnect": [],
      "on_mount": [],
      "props": {
        "breadcrumbItems": 0,
        "divider": 1,
        "image": 2,
      },
      "root": <div>
        <div
          class="cobalt svelte-1ehil4r"
        >
          <ul
            class="breadcrumb svelte-1ehil4r"
          >
            <li
              class="svelte-1ehil4r"
            >
               
              <a
                class="svelte-1ehil4r"
                data-testid="breadcrumbLink"
                href="/"
              >
                Dashboard
              </a>
               
            </li>
            <li
              class="svelte-1ehil4r"
            >
              <span
                class="svelte-1ehil4r"
              >
                /
              </span>
               
              
               
              <a
                class="svelte-1ehil4r"
                data-testid="breadcrumbLink"
                href="/reports"
              >
                Annual reports
              </a>
               
            </li>
            <li
              class="svelte-1ehil4r"
            >
              <span
                class="svelte-1ehil4r"
              >
                /
              </span>
               
              
               
              2019
               
            </li>
          </ul>
        </div>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="cobalt svelte-1ehil4r"
      >
        <ul
          class="breadcrumb svelte-1ehil4r"
        >
          <li
            class="svelte-1ehil4r"
          >
             
            <a
              class="svelte-1ehil4r"
              data-testid="breadcrumbLink"
              href="/"
            >
              Dashboard
            </a>
             
          </li>
          <li
            class="svelte-1ehil4r"
          >
            <span
              class="svelte-1ehil4r"
            >
              /
            </span>
             
            
             
            <a
              class="svelte-1ehil4r"
              data-testid="breadcrumbLink"
              href="/reports"
            >
              Annual reports
            </a>
             
          </li>
          <li
            class="svelte-1ehil4r"
          >
            <span
              class="svelte-1ehil4r"
            >
              /
            </span>
             
            
             
            2019
             
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
